hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieClusteringConfig.java

```diff
@@ -277,11 +266,9 @@ public class HoodieClusteringConfig extends HoodieConfig {
    */
   public static final ConfigProperty<String> LAYOUT_OPTIMIZE_SPATIAL_CURVE_BUILD_METHOD = ConfigProperty
       .key(LAYOUT_OPTIMIZE_PARAM_PREFIX + "curve.build.method")
-      .defaultValue("direct")
-      .sinceVersion("0.10.0")
-      .withDocumentation("Controls how data is sampled to build the space-filling curves. "
-          + "Two methods: \"direct\", \"sample\". The direct method is faster than the sampling, "
-          + "however sample method would produce a better data layout.");
+      .defaultValue(SpatialCurveCompositionStrategyType.DIRECT.name())
```

Comment:
similar here, we should make sure both lower- and upper-case names should work.

Reply:
Addressed